python/cudf/cudf/core/frame.py

```diff
@@ -238,6 +239,70 @@ def _quantiles(
         result._copy_categories(self)
         return result
 
+    def repeat(self, repeats, axis=None):
+        """Repeats elements consecutively
+
+        Parameters
+        ----------
+        repeats : int, array, numpy array, or Column
+            the number of times to repeat each element
+
+        Example
+        -------
+        >>> import cudf as cudf
+        >>> s = cudf.Series([0, 2]) # or DataFrame
+        >>> s
+        0    0
+        1    2
+        dtype: int64
+        >>> s.repeat([3, 4])
+        0    0
+        0    0
+        0    0
+        1    2
+        1    2
+        1    2
+        1    2
+        dtype: int64
+        >>> s.repeat(2)
+        0    0
+        0    0
+        1    2
+        1    2
+        dtype: int64
+        >>>
+        """
+        if axis is not None:
+            raise NotImplementedError(
+                "Only axis=`None` supported at this time."
+            )
+
+        return self._repeat(repeats)
+
+    def _repeat(self, count):
+        if is_scalar(count):
+            count = Scalar(count)
+        else:
+            count = as_column(count)
+
+        result = self.__class__._from_table(
+            libcudfxx.filling.repeat(self, count)
+        )
+
+        result._copy_categories(self)
+        return result
+
+    def _fill(self, fill_values, begin=0, end=-1, inplace=False):
+        col_and_fill = zip(self._columns, fill_values)
+
+        if not inplace:
+            data_columns = (c.fill(v, begin, end) for (c, v) in col_and_fill)
+            data = zip(self._column_names, data_columns)
+            return self.__class__._from_table(Frame(data, self._index))
+
+        for (c, v) in col_and_fill:
+            c.fill(v, begin, end, inplace=True)
```

Comment:
I imagine this doesn't work for StringColumns?

In that situation we can technically replace the Buffers underneath a StringColumn so that any other references to it are modified as well. Unsure what we should do here.

Reply:
`inplace` typically means "I'm trying not to allocate more memory".

throwing would make consuming code more complicated (compatibility/type/inplace check), so I'd want to avoid that.

Warning seems like the best approach here if `inplace` is used with non-fixed-width types. We can fall back to non-inplace operations

Reply:
> `inplace` typically means "I'm trying not to allocate more memory".

Agreed, but unfortunately Pandas has historically not always seen it this way where they do Buffer swaps themselves under the hood in a lot of situations. Additionally, their strings are really just a numpy array of `PyObject*` so they can easily just change the `PyObject*` objects inplace.

If you look at through the CategoricalColumn or StringColumn code you can see how we've built the `_mimic_inplace` function to handling this exact situation... unfortunately...

Reply:
This makes me sad.

Reply:
Oh, I still need to address this. oops.